protobuf/api/result.proto

```diff
@@ -0,0 +1,78 @@
+syntax = "proto3";
+
+import "gogo/protobuf/gogoproto/gogo.proto";
+import "protobuf/types/struct.proto";
+import "protobuf/types/result.proto";
+
+package mesg.api;
+option go_package = "api";
+
+// This is the API to interact with the Results.
+//
+// This API is a [gRPC](https://grpc.io/) API.
+//
+// The source file of this API is hosted on [GitHub](https://github.com/mesg-foundation/engine/blob/master/protobuf/api/result.proto).
+
+service Result {
+  // Create creates a single Execution specified in a request.
+  rpc Create(CreateResultRequest) returns (CreateResultResponse) {}
+
+  // Get returns a single Execution specified in a request.
+  rpc Get(GetResultRequest) returns (types.Result) {}
+
+  // List returns all Executions matching the criteria of the request.
+  rpc List(ListResultRequest) returns (ListResultResponse) {}
+
+  // Stream returns a stream of executions that satisfy criteria
+  // specified in a request.
+  rpc Stream(StreamResultRequest) returns (stream types.Result) {}
+}
+
+// CreateResultRequest defines request to create a single result.
+message CreateResultRequest {
+  // requestHash is hash of the associated execution.
+  bytes requestHash = 1 [
```

Comment:
```suggestion
  bytes executionHash = 1 [
```

We still have `Execution` type, not `Request` what do you thing?

Reply:
good catch